### PR TITLE
mksh: 57 -> 58

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/mksh/default.nix [hunk 1]
```diff
@@ -1,26 +1,29 @@
 { stdenv, fetchurl }:
 
+with stdenv.lib;
```
Comment (cole-h) [inline pkgs/shells/mksh/default.nix:3]: Not that it makes a difference, but why did you move this up here instead of keeping it down below, with what it's used for? It doesn't look like anything else needs `stdenv.lib` available...

Comment (AndersonTorres) [reply pkgs/shells/mksh/default.nix:3]: I prefer that way. So many times I needed all `stdenv.lib` functions readily available in other packages, then I started to declare it before everything.

Comment (cole-h) [reply pkgs/shells/mksh/default.nix:3]: IMHO, that's not a valid reason to move it. No reason to give it a larger scope than necessary.

If this package ever needs `lib` functions readily available, you can add `lib` to the arguments list and use `lib.optional`, for example, instead of `stdenv.lib.optional`.
